packages/core/src/lib/pseudopattern_hunter.js

```diff
@@ -60,10 +60,18 @@ pseudopattern_hunter.prototype.find_pseudopatterns = function(
       }
 
       //extend any existing data with variant data
-      variantFileData = _.merge(
+      variantFileData = _.mergeWith(
         {},
         currentPattern.jsonFileData,
-        variantFileData
+        variantFileData,
+        (objValue, srcValue) => {
+          if (
+            _.isArray(objValue) &&
+            patternlab.config.patternMergeVariantArrays
```

Comment:
If this is supposed the replicate the old behaviour, this needs to be !patternlab.config.patternMergeVariantArrays, since the old solution does _not_ merge arrays, but replaces the original array with the variant array.